File: Partner/partner/src/main/java/com/weeklycoffee/partner/subscribe/SubscribeDetail.java
package com.weeklycoffee.partner.subscribe;

import com.weeklycoffee.partner.product.Product;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@IdClass(SubscribeDetailId.class)
public class SubscribeDetail {

    @Id
    private long SubscribeId;
    @Id
    private int seq;

    @ManyToOne
    private Product product;

    private long partnerId;

    private String productName;
    private int productPrice;
    private int beanAmount;
    private int term;
    private int orderQuantity;
    private String groundPoint;
    private String processing;
    private String productImageId;
}

File: Partner/partnerclient/provider/modules/subscribeDetail.ts
import { createSlice } from "@reduxjs/toolkit";
import { ProductItem } from "./product";

export interface SubscribeDetail {
  subscribeId: number;

  seq: number;

  product: ProductItem[];

  partnerId: number;
  productName: string;
  productPrice: number;
  beanAmount: number;
  term: number;
  orderQuantity: number;
  groundPoint: string;
  productImageId: string;
}

const initialState: SubscribeDetail = {
  subscribeId: 0,
  seq: 0,
  product: [],
  partnerId: 0,
  productName: "",
  productPrice: 0,
  beanAmount: 0,
  term: 0,
  orderQuantity: 0,
  groundPoint: "",
  productImageId: "",
};

export const subscribeDetailSlice = createSlice({
  name: "subscribeDetail",
  initialState,
  reducers: {},
});

export const {} = subscribeDetailSlice.actions;

export default subscribeDetailSlice.reducer;
